music_assistant/server/providers/filesystem_local/helpers.py

```diff
@@ -79,7 +79,7 @@ def get_album_dir(track_dir: str, album_name: str) -> str | None:
         if compare_strings(album_name, dirname.split(" - ")[-1], False):
             # account for ArtistName - AlbumName format in the directory name
             return parentdir
-        if compare_strings(album_name, dirname.split("(")[0], False):
+        if compare_strings(album_name, dirname.split(" - ")[1].split("(")[0], False):
```

Comment:
this is dangerous - what if there is no "-" in the name ? crash
This needs either an additional guard or just use index -1 to grab the lst part

Reply:
thanks for catching that - it was meant to be `-1`. I have fixed it in the commit below and added some tests just to be sure.